### PR TITLE
Switch to JUnit version 5 for unit tests #4955

### PR DESCRIPTION
**Pull request by mclay**
- migrate unit tests in sdk/eventhubs/*

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java [hunk 1]
```diff
@@ -181,56 +179,60 @@ public void deserializeEventHubProperties() {
...
 
-        final Message message = Proton.message();
-        message.setBody(amqpValue);
+            final Message message = Proton.message();
+            message.setBody(amqpValue);
 
-        // Act
-        serializer.deserialize(message, EventHubProperties.class);
+            // Act
+            serializer.deserialize(message, EventHubProperties.class);
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java]: Only this line should be in the `assertThrows(() -> {})`, to scope down where we expect the error to be.

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java [hunk 2]
```diff
@@ -181,56 +179,60 @@ public void deserializeEventHubProperties() {
...
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java]: Only this line should be in the `assertThrows(() -> {})`, to scope down where we expect the error to be.

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java (hunk 1 above)
Comment (mclay) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java]: thanks for the review. changes are incorporated

File: sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java (hunk 2 above)
Comment (mclay) [reply sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubMessageSerializerTest.java]: thanks for the review. changes are incorporated

**Comment by JonathanGiles** [comment]
Thanks again @mclay!

**Pull request merged**